NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs

```diff
@@ -202,27 +190,99 @@ private IEnumerable<Entity> SpawnEntitiesStaticly(EntitySpawnPoint entitySpawnPo
 
         private IEnumerable<Entity> CreateEntityWithChildren(EntitySpawnPoint entitySpawnPoint, UnityEngine.Vector3 scale, TechType techType, int cellLevel, string classId, DeterministicBatchGenerator deterministicBatchGenerator)
         {
-            Entity spawnedEntity = new Entity(entitySpawnPoint.Position,
-                                              entitySpawnPoint.Rotation,
+            Entity spawnedEntity = new Entity(entitySpawnPoint.LocalPosition,
+                                              entitySpawnPoint.LocalRotation, 
                                               scale,
                                               techType,
                                               cellLevel,
                                               classId,
                                               true,
-                                              deterministicBatchGenerator.NextGuid());
+                                              deterministicBatchGenerator.NextGuid(),
+                                              true);
+
+            List<Entity> prefabs = GetEntitiesFromClassId(classId, spawnedEntity, deterministicBatchGenerator);
+
             yield return spawnedEntity;
 
             IEntityBootstrapper bootstrapper;
             if (customBootstrappersByTechType.TryGetValue(techType, out bootstrapper))
             {
                 bootstrapper.Prepare(spawnedEntity, deterministicBatchGenerator);
 
-                foreach(Entity childEntity in spawnedEntity.ChildEntities)
+                for (int i = 0; i < spawnedEntity.ChildEntities.Count - prefabs.Count; i++)
                 {
-                    yield return childEntity;
+                    Entity childEntity = spawnedEntity.ChildEntities[i + prefabs.Count];
```

Comment:
How does this work?

Also, it looks like we are no longer yielding these.

Reply:
ah right thats dead code... The client is Spawning ChildEntities so having them be returned while also existing as a child of an already returned entity will create bloat